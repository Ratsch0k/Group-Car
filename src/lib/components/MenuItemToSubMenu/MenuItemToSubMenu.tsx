import React from 'react';
import {
  MenuItemProps,
  MenuItem,
  makeStyles,
  createStyles,
  Theme,
} from '@material-ui/core';
import ArrowForwardIosIcon from '@material-ui/icons/ArrowForwardIos';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    arrowIcon: {
      fontSize: '1rem',
      position: 'absolute',
      right: theme.spacing(2),
    },
  }),
);

/**
 * Variant of the `MenuItem` which will add an arrow icon to the end of
 * the menu item to symbolize that this menu item will open an
 * a sub menu.
 * @param props Same as for `MenuItems`
 */
export const MenuItemToSubMenu: React.FC<MenuItemProps> = (props) => {
  const {children, ...rest} = props;

  const classes = useStyles();

  return (
    <MenuItem {...rest as unknown}>
      {children}
      <ArrowForwardIosIcon
        color='inherit'
        className={classes.arrowIcon}
      />
    </MenuItem>
  );
};

export default MenuItemToSubMenu;
