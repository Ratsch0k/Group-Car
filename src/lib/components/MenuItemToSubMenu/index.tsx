export * from './MenuItemToSubMenu';

export {default} from './MenuItemToSubMenu';
