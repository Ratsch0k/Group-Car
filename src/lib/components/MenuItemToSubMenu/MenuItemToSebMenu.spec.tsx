import { render } from "@testing-library/react";
import MenuItemToSubMenu from "./MenuItemToSubMenu";
import React from "react";

it('renders without crashing', () => {
  render(
    <MenuItemToSubMenu>
      TEST
    </MenuItemToSubMenu>
  )
});

it('matches snapshot', () => {
  const screen = render(
    <MenuItemToSubMenu>
      TEST
    </MenuItemToSubMenu>
  );

  expect(screen.baseElement).toMatchSnapshot();
});