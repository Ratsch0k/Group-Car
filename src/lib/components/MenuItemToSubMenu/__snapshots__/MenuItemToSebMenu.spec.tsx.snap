// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<body>
  <div>
    <li
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
      role="menuitem"
      tabindex="-1"
    >
      TEST
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-arrowIcon-2"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M5.88 4.12L13.76 12l-7.88 7.88L8 22l10-10L8 2z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root"
      />
    </li>
  </div>
</body>
`;
